### ISSUE TITLE
Nullpointer if catalog folder with custom types exits

### ISSUE THREAD

**Issue by holiday-sunrise** [mulesoft/mule-migration-assistant#461]
Try to migrate a API with custom data types

```
ERROR 2021-04-06 13:13:57,738 [main] com.mulesoft.tools.migration.engine.MigrationJob: Failed to apply task, rolling back and continuing with the next one.
com.mulesoft.tools.migration.exception.MigrationTaskException: Task execution exception. null
	at com.mulesoft.tools.migration.task.AbstractMigrationTask.execute(AbstractMigrationTask.java:94) ~[mule-migration-tool-api-1.2.0-SNAPSHOT.jar:?]
	at com.mulesoft.tools.migration.engine.MigrationJob.execute(MigrationJob.java:92) [mule-migration-tool-engine-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
	at com.mulesoft.tools.migration.MigrationRunner.main(MigrationRunner.java:82) [mule-migration-assistant-runner-1.2.0-SNAPSHOT.jar:1.2.0-SNAPSHOT]
Caused by: java.lang.NullPointerException
	at com.mulesoft.tools.migration.project.model.ApplicationModel.addNameSpace(ApplicationModel.java:195) ~[mule-migration-tool-api-1.2.0-SNAPSHOT.jar:?]
	at com.mulesoft.tools.migration.project.model.ApplicationModel.addNameSpace(ApplicationModel.java:176) ~[mule-migration-tool-api-1.2.0-SNAPSHOT.jar:?]
	at com.mulesoft.tools.migration.library.mule.steps.http.HttpNamespaceContribution.execute(HttpNamespaceContribution.java:27) ~[mule-migration-tool-library-1.2.0-SNAPSHOT.jar:?]
	at com.mulesoft.tools.migration.library.mule.steps.http.HttpNamespaceContribution.execute(HttpNamespaceContribution.java:18) ~[mule-migration-tool-library-1.2.0-SNAPSHOT.jar:?]
	at com.mulesoft.tools.migration.task.AbstractMigrationTask.lambda$execute$2(AbstractMigrationTask.java:68) ~[mule-migration-tool-api-1.2.0-SNAPSHOT.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1257) ~[?:1.8.0_212]
	at com.mulesoft.tools.migration.task.AbstractMigrationTask.execute(AbstractMigrationTask.java:68) ~[mule-migration-tool-api-1.2.0-SNAPSHOT.jar:?]
```

**Comment by holiday-sunrise** [comment]
fix could be.

ApplicationModel line 195

add Null check 

if (!schemaLocationAttribute.getValue().contains(namespace.getURI() + " ")
        && !schemaLocationAttribute.getValue().contains(schemaLocation)) {

**Comment by mlischetti** [comment]
Hi @holiday-sunrise 
Can you please try to isolate the issue in a simple Mule 3 app and add the example to the issue so that it can be reproduced?

**Comment by holiday-sunrise** [comment]
Fix the Nullpointer as requested. Why needs mulesoft alway example api´s. Each failure is fix is good enouch. Thats the way we clean code in germany. Thx

**Comment by afelisatti** [comment]
Hi, @holiday-sunrise! Examples allow us to include a test case for the issue. Then, we can make sure the changes fix the bug and no future changes break the scenario. These bugs are clearly coverage we are missing: adding the actual examples we've failed at is a great way to increase it. 
In some cases, examples also help us understand the use cases we are dealing with and inform improvements.
We've now included a template to help gather this information when reporting an issue and make the process easier. We hope this helps!
